flink-yarn-tests/src/test/java/org/apache/flink/yarn/testjob/YarnTestCacheJob.java

```diff
@@ -68,16 +70,21 @@ public static JobGraph getDistributedCacheJobGraph() {
 		@Override
 		public void open(Configuration config) throws IOException {
 			// access cached file via RuntimeContext and DistributedCache
-			File cacheFile = getRuntimeContext().getDistributedCache().getFile("cacheFile");
-			FileInputStream inputStream = new FileInputStream(cacheFile);
+			final File cacheFile = getRuntimeContext().getDistributedCache().getFile("cacheFile");
+			final FileInputStream inputStream = new FileInputStream(cacheFile);
 			properties = new Properties();
 			properties.load(inputStream);
 			checkArgument(properties.size() == 2, "The property file should contains 2 pair of key values");
+
+			final File testDirectory = getRuntimeContext().getDistributedCache().getFile(TEST_DIRECTORY_NAME);
+			if (!testDirectory.isDirectory()) {
+				throw new RuntimeException(String.format("%s is not a directory!", testDirectory.getAbsolutePath()));
+			}
```

Comment:
Do we have to test this feature via a Yarn test? Wouldn't it be easier to test the `FileCache` more targeted? For example, it could be done similar to `FileCacheDirectoriesTest#testDirectoryDownloadedFromBlob`. What we are interested in is the unpacking logic. The other part, that a registered cached filed is accessible on Yarn should already be tested by accessing the `cache.properties` file.

Reply:
I think we need the registering a directory as cache test for Yarn per-job. Because currently there is no existing test to cover the whole process. 
For Yarn session/Kubernetes session/Standalone, they share the same process. The directory needs to be compressed, uploaded to blob server via rest client, downloaded from blob, and then expanded. 
For Yarn per-job, it is different. The directory needs to be compressed, uploaded to hdfs, downloaded from hdfs, and then expanded. Add a file as cache could only cover part of the process. So i suggest to keep the test as now. Otherwise, when the whole process is broken, we will not know unless the users complain in the ML.

Also i will add a new unit test in `FileCacheDirectoriesTest#testDirectoryDownloadedFromDFS` to directly verify the changes.